libnavigation-base/api/current.txt

```diff
@@ -333,14 +337,17 @@ package com.mapbox.navigation.base.options {
 package com.mapbox.navigation.base.road.model {
 
   public final class Road {
-    method public java.util.List<com.mapbox.api.directions.v5.models.MapboxShield>? getMapboxShield();
-    method public String? getName();
-    method public String? getShieldName();
-    method public String? getShieldUrl();
-    property public final java.util.List<com.mapbox.api.directions.v5.models.MapboxShield>? mapboxShield;
-    property public final String? name;
-    property public final String? shieldName;
-    property public final String? shieldUrl;
```

Comment:
This is technically breaking SemVer. Are we ok with this as `RoadFactory` is `@ExperimentalMapboxNavigationAPI`? Should we also mark `Road` as `@ExperimentalMapboxNavigationAPI`? 🤔

Reply:
This is a class with an `internal` constructor. Since no one can access/create an instance of it directly, does it break SEMVER?

Reply:
Properties are not marked as `internal` and therefore they were originally available 😬

Reply:
Okay. I don't think we should mark it as `Experimental`. This object was made available in `v2.1` as 
```
class Road internal constructor(
    val name: String? = null,
    val shieldUrl: String? = null,
    val shieldName: String? = null,
    val mapboxShield: List<MapboxShield>? = null
) 
```

If we see, all the properties are nullable by default. What if we do this

```
class Road internal constructor(
    val components: List<RoadComponent>,
    val name: String? = null,
    val shieldUrl: String? = null,
    val shieldName: String? = null,
    val mapboxShield: List<MapboxShield>? = null
) 
```

@Guardiola31337

Reply:
That should work. You can also remove them from constructor and probably mark these properties as deprecated:
```kotlin
class Road internal constructor(
    val components: List<RoadComponent>,
) {
    @Deprecated val name: String? = null
    @Deprecated val shieldUrl: String? = null
    @Deprecated val shieldName: String? = null
    @Deprecated val mapboxShield: List<MapboxShield>? = null
}
```